merge spec 0.7.6

estebanlm opened the pull request:
https://github.com/pharo-spec/Spec/releases/tag/v0.7.6

- several enhancements and bugfixes to `SpPopoverPresenter`.
- in `SpStyle`, introduced the concept of environment variables. Now we can define EnvirontmentFont and EnvironmentColor to reference environment defined variables.
- fixed usage of minWidth/minHeight in styles.
- fix https://github.com/pharo-spec/Spec/issues/960
- fixed in how `SpBoxLayout` calculates widths and heights, allowing the toolbar to take its real place.
- fixed alignments in `SpBoxLayout` (hAlign* and vAlign* options now work properly).
- added `SpToolbarPopoverButtonPresenter`.
- `SpMillerColumnPresenter` now understands `removeAllFrom:`
- in tables, fixed the usage of showColumnHeaders (now it show headers even if headers are nil).
- in tables, fixed image columns with fixed widths (now calculate correctly the width).
- `SpCodePresenter` commands  use new selector extractor.

estebanlm committed:
merge version 0.7.5

estebanlm committed:
fix tests

estebanlm committed:
fix merging and font size calculous

estebanlm committed:
fix icon button building (do not add a morph even if text is empty)
fix test parse font

estebanlm committed:
new fix to mergeWith: fonts
fix calculous of box extents (taking into account the border width)

estebanlm committed:
yet another small fix to font variables

estebanlm committed:
and another (final?)

estebanlm committed:
implement expandAll for trees

estebanlm committed:
fixed action bars with new SpToolbarMorph changes

Ducasse commented:
Esteban there are some broken tests

windows-64 / Tests-windows-64 / testWorkerPoolDoesNotExceedPoolSizeWhenSchedulingTasksInParallel – Windows64.TaskIt.Tests.Worker.TKTCommonQueueWorkerPoolTest
<1s
Existing failures - 10
unix-64 / Tests-unix-64 / testInstallSelectedMetalinkActionButton – Unix64.Calypso.SystemPlugins.Reflectivity.Browser.Tests.Metalinks.ClyInstallMetaLinkPresenterTest
<1s
unix-64 / Tests-unix-64 / testUninstallSelectedMetalinkActionButton – Unix64.Calypso.SystemPlugins.Reflectivity.Browser.Tests.Metalinks.ClyInstallMetaLinkPresenterTest
<1s
unix-64 / Tests-unix-64 / testLocalMethodsOfTheClassShouldNotBeRepeatedInItsTraits – Unix64.ReleaseTests.Release.ReleaseTest

estebanlm commented:
But this is related to Spec2?

Esteban
On Nov 11 2020, at 4:13 pm, StéphaneDucasse <notifications@github.com> wrote:
>
>
> Esteban there are some broken tests
> windows-64 / Tests-windows-64 / testWorkerPoolDoesNotExceedPoolSizeWhenSchedulingTasksInParallel – Windows64.TaskIt.Tests.Worker.TKTCommonQueueWorkerPoolTest
> <1s
> Existing failures - 10
> unix-64 / Tests-unix-64 / testInstallSelectedMetalinkActionButton – Unix64.Calypso.SystemPlugins.Reflectivity.Browser.Tests.Metalinks.ClyInstallMetaLinkPresenterTest
> <1s
> unix-64 / Tests-unix-64 / testUninstallSelectedMetalinkActionButton – Unix64.Calypso.SystemPlugins.Reflectivity.Browser.Tests.Metalinks.ClyInstallMetaLinkPresenterTest
> <1s
> unix-64 / Tests-unix-64 / testLocalMethodsOfTheClassShouldNotBeRepeatedInItsTraits – Unix64.ReleaseTests.Release.ReleaseTest
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub (https://github.com/pharo-project/pharo/pull/7713#issuecomment-725474978), or unsubscribe (https://github.com/notifications/unsubscribe-auth/AAD5MXX6KUDUNXE2OEM5HALSPKSZ3ANCNFSM4TQP5JYQ).
>

estebanlm committed:
- remove duplicated methods
- fix stupid error reported by @martinmcclure ;)
- ensure newShownPages has a minimum size of one (fast draggings of edges was causing some errors)

estebanlm committed:
fix click simulation (because now is using wrong execution method)

estebanlm committed:
using click instead execute

MarcusDenker commented:
all tests are green (but the Zinc test on Win, but that is not related)
